models/purchase.py

```diff
@@ -111,6 +113,17 @@ def __repr__(self):
     def is_transferable(self):
         return self.product.get_attribute("is_transferable")
 
+    def is_refundable(self, override_refund_state_machine=False) -> bool:
+        return (
+            (self.is_paid_for is True)
+            and not self.is_transferred
+            and self.payment.is_refundable(override_refund_state_machine)
+        )
```

Comment:
```suggestion
        return (
            self.is_paid_for
            and not self.is_transferred
            and self.payment.is_refundable(override_refund_state_machine)
        )
```

(seems consistent with elsewhere in the codebase)

Reply:
this change is purely there so that mypy is happy (otherwise it sees `is_paid_for` as a column type rather than a boolean

Reply:
Gotcha; frustrating, but fair enough (I had a quick look at adjusting the definition / `column_property` to resolve that, and couldn't immediately find a way to do that; at least not without [making it a Python `@property`](https://docs.sqlalchemy.org/en/20/orm/mapped_sql_expr.html#using-a-plain-descriptor), and apparently that'd incur performance degradation).